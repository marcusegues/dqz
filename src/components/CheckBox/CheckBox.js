// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';

import { MAIN_RED, GREY } from '../../styles/colors';
import { moderateScale } from '../../styles/Scaling';

export const CheckBox = ({ checked }: { checked: boolean }) => (
  <View style={{ flexDirection: 'row', alignItems: 'center' }}>
    {checked ? (
      <MaterialIcons
        name="check-circle"
        size={moderateScale(35)}
        color={MAIN_RED}
      />
    ) : (
      <MaterialIcons
        name="radio-button-unchecked"
        size={moderateScale(35)}
        color={GREY}
      />
    )}
  </View>
);
