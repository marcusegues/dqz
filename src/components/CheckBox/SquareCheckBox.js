// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import { moderateScale } from '../../styles/Scaling';

export const SquareCheckBox = ({ checked }: { checked: boolean }) => (
  <View style={{ flexDirection: 'row', alignItems: 'center' }}>
    {checked ? (
      <MaterialCommunityIcons
        name="checkbox-marked"
        size={moderateScale(25)}
        color="#4A4A4A"
      />
    ) : (
      <MaterialCommunityIcons
        name="checkbox-blank-outline"
        size={moderateScale(25)}
        color="#4A4A4A"
      />
    )}
  </View>
);
