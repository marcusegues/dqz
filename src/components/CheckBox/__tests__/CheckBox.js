import React from 'react';
import renderer from 'react-test-renderer';
import { CheckBox } from '../CheckBox';

describe('CheckBox', () => {
  test('renders according to snapshot unchecked', () => {
    const component = renderer.create(<CheckBox checked={false} />).toJSON();
    expect(component).toMatchSnapshot();
  });
  test('renders according to snapshot checked', () => {
    const component = renderer.create(<CheckBox checked />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
