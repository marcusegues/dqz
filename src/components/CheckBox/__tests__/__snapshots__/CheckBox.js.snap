// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox renders according to snapshot checked 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#DC0018",
          "fontSize": 55,
        },
        undefined,
        Object {
          "fontFamily": "material",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    î¡¬
  </Text>
</View>
`;

exports[`CheckBox renders according to snapshot unchecked 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;
